### PR TITLE
fix: avoid nil-pointer panic when checking the frontend IP configuration

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind bug

**What this PR does / why we need it**:

fix: avoid nil-pointer panic when checking the frontend IP configuration. This is needed when the fip config is not bound to an public IP.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```
fix: avoid nil-pointer panic when checking the frontend IP configuration
```

### CODE REVIEW

**Comment by nilo19** [comment]
fix in k/k: https://github.com/kubernetes/kubernetes/pull/101739

File: pkg/provider/azure_standard.go [hunk 1]
```diff
@@ -353,7 +353,7 @@ func (az *Cloud) serviceOwnsFrontendIP(fip network.FrontendIPConfiguration, serv
 		}
 
 		if pip != nil && pip.ID != nil && pip.PublicIPAddressPropertiesFormat != nil && pip.IPAddress != nil {
-			if strings.EqualFold(*pip.ID, *fip.PublicIPAddress.ID) {
+			if strings.EqualFold(to.String(pip.ID), to.String(fip.PublicIPAddress.ID)) {
```
Comment (feiskyer) [inline pkg/provider/azure_standard.go:356]: also check `fip.FrontendIPConfigurationPropertiesFormat != nil && fip.FrontendIPConfigurationPropertiesFormat.PublicIPAddress != nil`?

Comment (nilo19) [reply pkg/provider/azure_standard.go:361]: done

File: pkg/provider/azure_standard.go [hunk 2]
```diff
@@ -352,8 +352,13 @@ func (az *Cloud) serviceOwnsFrontendIP(fip network.FrontendIPConfiguration, serv
...
 		}
 
-		if pip != nil && pip.ID != nil && pip.PublicIPAddressPropertiesFormat != nil && pip.IPAddress != nil {
-			if strings.EqualFold(*pip.ID, *fip.PublicIPAddress.ID) {
+		if pip != nil &&
+			pip.ID != nil &&
+			pip.PublicIPAddressPropertiesFormat != nil &&
+			pip.IPAddress != nil &&
+			fip.FrontendIPConfigurationPropertiesFormat != nil &&
+			fip.PublicIPAddress != nil {
```
Comment (feiskyer) [inline pkg/provider/azure_standard.go:360]: better to use `fip.FrontendIPConfigurationPropertiesFormat.PublicIPAddress` here for clear

**Pull request merged**